### PR TITLE
Avoid `TypeError` when params contain a key without a value on Ruby < 2.4

### PR DESCRIPTION
**Pull request by segiddins**
Fixes https://github.com/sinatra/sinatra/issues/1514

### CODE REVIEW

**Comment by segiddins** [comment]
This ought to go green when rebased atop https://github.com/sinatra/sinatra/pull/1513

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+## Unreleased
+
+* Avoid `TypeError` when params contain a key without a value on Ruby < 2.4 [#1516](https://github.com/sinatra/sinatra/pull/1516) by Samuel Giddins
+
```
[the remark this answers is not in the document]
Comment (segiddins) [reply CHANGELOG.md]: The contributing.md said it made it easier to do releases when PRs included changelog entries 😉

[the remark this answers is not in the document]
Comment (namusyaka) [reply CHANGELOG.md]: The changelog entry will automatically be generated by the release script: #1515
contributing.md will also updated at the same time.

[the remark this answers is not in the document]
Comment (namusyaka) [reply CHANGELOG.md]: @segiddins Yes, I apologize if confusing.
But as I said earlier, the contents of contributing.md will be updated soon.
Do you want to keep your own changelog entry so much?
If not, please remove the entry from changelog.

[the remark this answers is not in the document]
Comment (junaruga) [reply CHANGELOG.md]: This PR is not #1516 but #1517. That confused me a little :)
> [#1516](https://github.com/sinatra/sinatra/pull/1516)

[the remark this answers is not in the document]
Comment (segiddins) [reply CHANGELOG.md]: agh, I was trying to figure out which PR it would be before opening it

**Comment by RaVbaker** [comment]
Just realized it doesn't cover one other case. When the parameter is `true` Because in Ruby <=2.3 also making a `dup` on `true` is prohibited. While this PR will cover the `false` and `nil` cases. Yet not the `true`. 

```> true.dup
TypeError: can't dup TrueClass```

while in >=2.4

```> true.dup
=> true```

And yet in JSON body payloads there can be a `true`, `false` or `null/nil`. Which are not dup-able in older Rubies. @segiddins @namusyaka

**Comment by dentarg** [comment]
@RaVbaker do you have a repro for the issue? I think it is better to open up a new issue than commenting here

**Comment by namusyaka** [comment]
@RaVbaker Good catch, I'll work on the issue soon

**Pull request merged**